src/v/kafka/protocol/wire.h

```diff
@@ -72,6 +73,12 @@ class decoder {
         auto [i, _] = _parser.read_unsigned_varint();
         return i;
     }
+    double read_float64() {
+        static_assert(
+          std::numeric_limits<float64_t>::is_iec559,
+          "Kafka float64 type should be IEEE 754 (IEC599)");
+        return std::bit_cast<float64_t>(read_int64());
```

Comment:
I'd stick to double. But if you want to use float64_t in c++, then shouldn't you return that type too for consistency?